pkg/oktaapi/client_wrapper.go

```diff
@@ -100,13 +106,18 @@ func (cw *clientWrapper) SearchByName(ctx context.Context, searchTerm string) ([
 		return nil, err
 	}
 
-	users := make([]*models.UserInfo, len(searchedUsers))
-	for idx, user := range searchedUsers {
+	users := []*models.UserInfo{}
+	for _, user := range searchedUsers {
 		profile, err := cw.parseOktaProfileResponse(user.Profile)
 		if err != nil {
 			return nil, err
 		}
-		users[idx] = profile
+
+		// If we find EUA users that have logins longer than 4 characters, they're a test user (don't add them to the array)
+		if len(profile.Login) > 4 {
```

Comment:
@ClayBenson94 , can you speak to this approach? This will remove all the test accounts, but it also excludes all non - eua users from search. I _think_ that makes sense with what we were already doing with this search.

I wonder if it makes sense to update the name of the search to note we are only searching for EUA users?

Reply:
Thoughts on this approach?
```go
	users := []*models.UserInfo{}
	for _, user := range searchedUsers {
		profile, err := cw.parseOktaProfileResponse(user.Profile)
		if err != nil {
			return nil, err
		}

		// If we find EUA users that have logins longer than 4 characters, they're a test user (don't add them to the array)
		if (profile.SourceType == euaSourceType || profile.SourceType == euaADSourceType) && len(profile.Login) > 4 {
			continue
		}
		users = append(users, profile.toUserInfo())
	}
```

Reply:
This way the function stays agnostic to what we're searching, but it only filters out the EUA users? That way if we later add MAC/IDM User searching, this doesn't conflict with that

Reply:
c762b413aceefc2fd5a4b5b4bd85be04523afdd6